### DISCUSSION TITLE
Cypress seems to ignore the `specPattern` option in cypress.config.js

### DISCUSSION THREAD

**Discussion by paul9115** [cypress-io/cypress#26097] [Questions and Help]
I'm seeing an issue when setting up my cypress tests and attempting to configure spec patterns (I'm using the cucumber plugin and want to limit spec files to .feature files. 
In my config I have the following:
```
e2e: {
        specPattern: 'foo/bar/*.cy.{feature,features,api.js}',
        baseUrl: '<URL>',
        setupNodeEvents(on, config) {
            on('file:preprocessor', cucumber());
            ...
```
However, when running Cypress and checking the specs everything is listed and the options show that `specPattern` is set to `**/tests/**/*`  
No matter what I set this glob to the default value is used instead. 
Other options such as `baseUrl` are working as intended and the config in settings shows specPattern is being set from the config

**Comment by portyu9** [comment]
Hi @paul9115 

Cypress does support `e2e.specPattern`, so if the **Settings/Options** view resolves it to `**/tests/**/*`, something in the configuration/plugin setup is likely replacing or modifying the value.

The Cucumber setup in the example is also using the old `cypress-cucumber-preprocessor` integration. For current Cypress, use the maintained `@badeball/cypress-cucumber-preprocessor` and make sure `setupNodeEvents()` returns the resulting config.

For example:

```ts
import { defineConfig } from 'cypress'
import createBundler from '@bahmutov/cypress-esbuild-preprocessor'
import { addCucumberPreprocessorPlugin } from '@badeball/cypress-cucumber-preprocessor'
import { createEsbuildPlugin } from '@badeball/cypress-cucumber-preprocessor/esbuild'

export default defineConfig({
  e2e: {
    specPattern: 'foo/bar/**/*.feature',

    async setupNodeEvents(on, config) {
      await addCucumberPreprocessorPlugin(on, config)

      on(
        'file:preprocessor',
        createBundler({
          plugins: [createEsbuildPlugin(config)],
        })
      )

      return config
    },
  },
})
```

If you need to stay on the older plugin temporarily, log `config.specPattern` at the beginning and end of `setupNodeEvents()` and check every plugin/config helper that can return or replace Cypress configuration.

The important clue here is that Cypress's resolved configuration shows `**/tests/**/*`; that means the value Cypress ultimately received is different from the `specPattern` shown in the snippet rather than Cypress simply ignoring `specPattern`.

Also restart Cypress after changing the configuration so you are testing a fresh config load.

Please mark this answer as accepted if it helped, thank you.
